packages/react/src/use-chat-2.ts

```diff
@@ -135,50 +98,17 @@ export type UseChatOptions2<
    * Initial input of the chat.
    */
   initialInput?: string;
-
-  /**
-  Optional callback function that is invoked when a tool call is received.
-  Intended for automatic client-side tool execution.
-
-  You can optionally return a result for the tool call,
-  either synchronously or asynchronously.
-     */
-  onToolCall?: ({
-    toolCall,
-  }: {
-    toolCall: ToolCall<string, unknown>;
-  }) => void | Promise<unknown> | unknown;
-
-  /**
-   * Optional callback function that is called when the assistant message is finished streaming.
-   *
-   * @param message The message that was streamed.
-   */
-  onFinish?: (options: {
-    message: UIMessage<MESSAGE_METADATA, InferUIDataParts<DATA_TYPE_SCHEMAS>>;
-  }) => void;
-
-  /**
-   * Callback function to be called when an error is encountered.
-   */
-  onError?: (error: Error) => void;
-
-  /**
-   * A way to provide a function that is going to be used for ids for messages and the chat.
-   * If not provided the default AI SDK `generateId` is used.
-   */
-  generateId?: IdGenerator;
-};
```

Comment:
Also pushed back onto `AbstractChat` -- there was no benefit to having these callbacks live at this layer since they're passed to every single network method anyway. Lets us delete a lot of code.